controllers/iamauthpolicy_controller.go

```diff
@@ -68,8 +78,8 @@ func (c *IAMAuthPolicyController) Reconcile(ctx context.Context, req ctrl.Reques
 
 	latticeResourceId, err := reconcileFunc(ctx, k8sPolicy)
 	if err != nil {
-		if services.IsNotFoundError(err) {
-			c.log.Infof("reconcile error, retry in 30sec: %s", err)
+		if !isDelete && services.IsNotFoundError(err) {
```

Comment:
What this condition used for `!isDelete`?

Reply:
Thanks for pointing again, logic is wrong in the code. Intention was:
If it's policy delete request and SN or Svc not found, dont throw error, just remove finalizer and we done.

Reply:
updated condition

Reply:
we should start doing unit tests in controllers :D

Reply:
Why I think old code is correct... `!isDelete && services.IsNotFoundError(err)`

Reply:
if it's policy delete and svc not found ignore, otherwise throw -> (isDelete && notFound) -> ignore, !(isDelete && notFound) -> throw